opendc-workflow/opendc-workflow-service/src/main/kotlin/org/opendc/workflow/service/internal/StateHelpers.kt

```diff
@@ -0,0 +1,81 @@
+package org.opendc.workflow.service.internal
+
+import org.opendc.workflow.api.WORKFLOW_TASK_DEADLINE
+import org.opendc.workflow.api.WORKFLOW_TASK_MINIMAL_START_TIME
+import org.opendc.workflow.api.WORKFLOW_TASK_SLACK
+import java.util.UUID
+import kotlin.collections.HashMap
+import kotlin.collections.HashSet
+import kotlin.math.max
+
+public fun JobState.computeSlack() {
+    this.computeMinimalStartTimes()
+    for (t in this.tasks) {
+        if (t.dependencies.isEmpty())  {
+            t.task.metadata[WORKFLOW_TASK_SLACK] = 0L
+        } else {
+            val cmin = t.dependencies.minWith(Comparator.comparingLong {it.task.metadata[WORKFLOW_TASK_SLACK] as Long})
```

Comment:
After splitting it out into the following lines, I still get a runtime exception from casting a null value to long.

`    for (t in this.tasks) {
        if (t.dependencies.isEmpty())  {
            t.task.metadata[WORKFLOW_TASK_SLACK] = 0L
        } else {
            val cmin = t.dependencies.minWith(Comparator.comparingLong {t.task.metadata[WORKFLOW_TASK_SLACK]?: 0L as Long})
            val taskslack: Long = cmin.task.metadata[WORKFLOW_TASK_SLACK]?: 0L
            val minstarttime: Long = t.task.metadata[WORKFLOW_TASK_MINIMAL_START_TIME]?: 0L // exception here
            val taskdeadline: Long = t.task.metadata[WORKFLOW_TASK_DEADLINE]?: 0L
            t.task.metadata[WORKFLOW_TASK_SLACK] = taskslack - minstarttime + taskdeadline
        }
    }
`